### PR TITLE
ROC-5854: Refactor and code cleanup to remove duplicate

### PR DESCRIPTION
**Pull request by sabahirfan**
### JIRA ###
[ROC-5854](https://tools.hmcts.net/jira/browse/ROC-5854)

### Change description ###
Refactor and code cleanup to remove duplicate

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseApi.java [hunk 1]
```diff
@@ -162,7 +162,7 @@ private void linkToCaseWithoutRevoking(
         User anonymousCaseWorker,
         String caseId
     ) {
-        LOGGER.debug("Granting access to case {} for defendant {}", caseId, defendantId);
+        LOGGER.debug("Granting access extractClaim case {} for defendant {}", caseId, defendantId);
```
Comment (markry-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseApi.java]: search replace error me thinks 🙈

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/LinkDefendantToClaimWithWithCoreCaseDataTest.java [hunk 2]
```diff
@@ -54,7 +54,7 @@
     private static final String ANONYMOUS_USER_ID = "3";
 
     @Autowired
-    private CCDCaseDataToClaim ccdCaseDataToClaim;
+    private CaseDetailsConverter ccdCaseDataToClaim;
```
Comment (markry-kainos) [inline src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/LinkDefendantToClaimWithWithCoreCaseDataTest.java]: pls rename var too

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java [hunk 3]
```diff
@@ -99,7 +99,7 @@ public ClaimService(
         MoreTimeRequestRule moreTimeRequestRule,
         EventProducer eventProducer,
         AppInsights appInsights,
-        CCDCaseDataToClaim ccdCaseDataToClaim,
+        CaseDetailsConverter ccdCaseDataToClaim,
```
Comment (markry-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java]: var name too

File: src/test/java/uk/gov/hmcts/cmc/claimstore/utils/CaseDetailsConverterTest.java [hunk 4]
```diff
@@ -20,30 +20,30 @@
 @SpringBootTest
 @ContextConfiguration(classes = CCDMapperConfig.class)
 @RunWith(SpringJUnit4ClassRunner.class)
-public class CCDCaseDataToClaimTest {
+public class CaseDetailsConverterTest {
 
-    private CCDCaseDataToClaim ccdCaseDataToClaim;
+    private CaseDetailsConverter ccdCaseDataToClaim;
```
Comment (markry-kainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/utils/CaseDetailsConverterTest.java]: var name

File: src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseApi.java (hunk 1 above)
Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseApi.java]: right. 👍 done

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/LinkDefendantToClaimWithWithCoreCaseDataTest.java (hunk 2 above)
Comment (sabahirfan) [reply src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/LinkDefendantToClaimWithWithCoreCaseDataTest.java]: done

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java (hunk 3 above)
Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java]: done

File: src/test/java/uk/gov/hmcts/cmc/claimstore/utils/CaseDetailsConverterTest.java (hunk 4 above)
Comment (sabahirfan) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/utils/CaseDetailsConverterTest.java]: done

**Pull request merged**